### ISSUE TITLE
Repeated results in RD_CLOUD section. 

### ISSUE THREAD

**Issue by kratoz29** [kodifitzwell/repo#9]
I have noticed that when I set up to scrape my RD cloud several scrapped items show up twice in said section... Is there any reason why that happens?

Also it would be nice to have a different coloring option for the RD_CLOUD field :)

**Comment by kodifitzwell** [comment]
when a particular item is in "cloud storage" and "history" it shows as two links when "search RD cloud" is enabled.  one has been resolved and can play, one has to be resolved first before play.

it's a big guess.  I ran a quick test, I didn't really look at the returns from the api calls.

>Also it would be nice to have a different coloring option for the RD_CLOUD field :)

not sure what you mean.  coloring is based addon settings: quality, provider, or scraper type.  I prefer the setting to sort it to the top as well.

**Comment by kratoz29** [comment]
> when a particular item is in "cloud storage" and "history" it shows as two links when "search RD cloud" is enabled. one has been resolved and can play, one has to be resolved first before play.

Huh, I am afraid I don't follow, what is this "history" you are talking about? I am pretty sure any of the repeated links playback quite quickly. 

> not sure what you mean. coloring is based addon settings: quality, provider, or scraper type. I prefer the setting to sort it to the top as well.

I have it set up to put my RD CLOUD results at the top, but they are indistinguishable from the other 4K results as they both use the same color, and usually the top results are also the same ones that would should in the RD CLOUD results. 

You can tweak the color highlights for each of the four qualities, but it would be nice to have a fifth option to specify top results are RD CLOUD ones, so you could quickly differentiate them from the others as they would not have the same magenta color (the default one for 4K).

**Comment by kodifitzwell** [comment]
>Huh, I am afraid I don't follow, what is this "history" you are talking about?

have you never visited the My Services menu?

![capture1](https://github.com/user-attachments/assets/759a6f9d-1941-4b83-bb0c-5c43ac278c63)

RD_CLOUD is quite prominent in two places, one being the same place you differentiate the source (torrentio, bitsearch, etc).

![capture2](https://github.com/user-attachments/assets/c8804ffd-a7be-48d4-b01d-449ec0e89689)

if it is its own color, then one would need to focus on the same area where RD_CLOUD is also distinguished to determine the quality.  so it seems, either way one needs to look in the general area.  I feel it is most logical that it matches the resolution sort color since I have results sorted by quality.

**Comment by kratoz29** [comment]
> have you never visited the My Services menu?

I just did... So is this the reason why the results are repeated? I don't understand the difference, AFAIK any resource used from RD should be automatically added to your RD account? I mean this has been the behavior I have seen with several "RD clients". 

> if it is its own color, then one would need to focus on the same area where RD_CLOUD is also distinguished to determine the quality. so it seems, either way one needs to look in the general area. I feel it is most logical that it matches the resolution sort color since I have results sorted by quality.

Maybe a good indicator would be to fill the left area with the same color then? I mean just for having a quick glance to differentiate them.

**Comment by kodifitzwell** [comment]
> So is this the reason why the results are repeated? I don't understand the difference, AFAIK any resource used from RD should be automatically added to your RD account? I mean this has been the behavior I have seen with several "RD clients".

as I said in my first reply, most likely it is.

"cloud storage" corresponds to the "torrents" section when you log in to your RD account.

![capture1](https://github.com/user-attachments/assets/98a26add-c332-4404-ada1-ce3ab633a134)

"history" corresponds to the "downloads history".  anything you start to stream/download (get resolved) is added to history.

![capture2](https://github.com/user-attachments/assets/b74657e3-597d-4111-b3b7-7672d88c2349)

  you can add to cloud without starting stream/download and vice versa.  there is a setting in the addon to enable/disable "add resolved torrents to cloud".  when this is disabled, the addons deletes it after resolution so it only shows in your "history" and does not clutter your "cloud storage".

> Maybe a good indicator would be to fill the left area with the same color then? I mean just for having a quick glance to differentiate them.

I'm quite content with the logic the original dev has applied here.  rather than color, RD_CLOUD is used to difference with a quick glance.

**Comment by kratoz29** [comment]
>  there is a setting in the addon to enable/disable "add resolved torrents to cloud". when this is disabled, the addons deletes it after resolution so it only shows in your "history" and does not clutter your "cloud storage".


Hmm, I had that setting enabled, I just disabled it and deleted the add-on's cache to see if repeated links showed up, and it seems like they are still showing.

I tried with another movie (that ain't in my "In Progress" section) and the previously scrapped link ain't showing up 😅

> I'm quite content with the logic the original dev has applied here. rather than color, RD_CLOUD is used to difference with a quick glance.

Well yeah, having those results already on top is so handy... But personally I confuse them with the next top results lol, and if we take in consideration that I already see repeated results in the "not full scrapping" section yeah I feel like I'm seeing the same results over and over.

Anyway this was just a suggestion, what can I do to stop seeing results twice in the RD CLOUD scrap?

**Comment by kodifitzwell** [comment]
>Hmm, I had that setting enabled, I just disabled it and deleted the add-on's cache to see if repeated links showed up, and it seems like they are still showing.

where do you think the addons local cache gets its data from?  probably makes more sense do delete it from where the local cache pulls from.  😅

>I tried with another movie (that ain't in my "In Progress" section) and the previously scrapped link ain't showing up 😅

the cloud scraper is basically a filename filter.  depending on how the file is named there could be many reasons it does not match or falsely matches the dozens of filenames already in your "cloud storage" and "history".  show me your better coded algo and I'll replace the current one.  😅

>what can I do to stop seeing results twice in the RD CLOUD scrap?

disable cloud scraping and you will never see two cloud scrape results again.  😅

**Comment by kratoz29** [comment]
> where do you think the addons local cache gets its data from? probably makes more sense do delete it from where the local cache pulls from. 😅

Oh, I have tried that before, but I have never seen any impact with POV results (used the unchained app for Android) I might try it again. 

>  show me your better coded algo and I'll replace the current one. 😅

Huh?

> disable cloud scraping and you will never see two cloud scrape results again. 

lol, fair enough, is not like POV is slow by any means... Is just that I really use that feature to filter out undesired results (like films that does not have Spanish language for example).

**Comment by kodifitzwell** [comment]
>Huh?

you said:

>I tried with another movie (that ain't in my "In Progress" section) and the previously scrapped link ain't showing up 😅

so I asked to see your python code that matches the api result from RD 100% of the time so I can replace the current addon code.  😅

>Is just that I really use that feature to filter out undesired results (like films that does not have Spanish language for example).

so the cloud results are the desired or less desired results?

**Comment by kratoz29** [comment]
> o see your python code that matches the api result from RD 100% of the time so I can replace the current addon code. 😅

Oh gotcha, trust me, if I knew python I'd be more than glad to help you on this, but I am afraid I do not know... I want to learn it though, for personal stuff.

What I can tell you is that the scrapped link that I just tried in POV (for the movie Twisters, weird as I can see one of those results in your previous image) is still not showing up there... But it does in the RD CLOUD section of Fen Light... So maybe it is worth checking it there? 

> so the cloud results are the desired or less desired results?

The cloud results are the desired ones, I normally don't finish a movie when I start it, so I usually go back to the RD CLOUD scrapped results and it is nice to have the "best link" for me already there, pretty much as Stremio handles it (it always remembers the last played link). 

I removed all the clutter with the Unchained app, and disabled the setting for "add resolved torrents to cloud" and tried some links and so far all good, no repeated stuff.
